runners/spark/src/main/java/org/apache/beam/runners/spark/translation/EvaluationContext.java

```diff
@@ -195,23 +193,24 @@ public void computeOutputs() {
     throw new IllegalStateException("Cannot resolve un-known PObject: " + value);
   }
 
-  @Override
-  public <T> T getAggregatorValue(String named, Class<T> resultType) {
-    return runtime.getAggregatorValue(AccumulatorSingleton.getInstance(jsc), named, resultType);
-  }
-
-  @Override
   public <T> AggregatorValues<T> getAggregatorValues(Aggregator<?, T> aggregator)
       throws AggregatorRetrievalException {
     return runtime.getAggregatorValues(AccumulatorSingleton.getInstance(jsc), aggregator);
   }
 
-  @Override
-  public MetricResults metrics() {
-    throw new UnsupportedOperationException("The SparkRunner does not currently support metrics.");
+  public <T> T getAggregatorValue(String named, Class<T> resultType) {
```

Comment:
Well, if it's between delegating `getAggregatorValue/s` or providing `SparkRuntimeContext` - I think that it is OK to expect a context (`EvaluationContext`) to provide stuff, rather than delegate to it. WDYT ?